### PR TITLE
Error handling for crashes in replication manager or worker threads

### PR DESCRIPTION
**Pull request by prakashngit**
Currently, it is possible that the application hangs if the replication manager or worker threads crash. This PR partially addresses this problem. A full solution will be provided soon. 

Signed-off-by: prakashngit <prakash.narayana.moorthy@intel.com>

### CODE REVIEW

File: python/pdo/contract/replication.py [hunk 1]
```diff
@@ -111,26 +124,38 @@ def __replication_manager__():
...
+                        __services_to_ignore__.add(service_id)
 
             # get the set of services to use with this task
             ids_of_services_to_use = replication_request.service_ids - __services_to_ignore__
 
             #check that there are enough services for replication, else add to the set of failed tasks and go to the next task
             if len(ids_of_services_to_use) < replication_request.num_provable_replicas:
-                logger.error("Replication failed for request number %d. Not enough reliable storage services.", request_id)
+                logger.error("Replication failed for request number %d. Either not enough reliable storage services or too many worker threads crashed." \
+                    , request_id)
```
Comment (cmickeyb) [inline python/pdo/contract/replication.py]: just put the comma on the previous line then you don't need the "\"

File: python/pdo/contract/replication.py [hunk 2]
```diff
@@ -111,26 +124,38 @@ def __replication_manager__():
...
                 continue
 
-            #ensure that the worker threads and queues are in place for services associated with this replication request
+            #ensure that the worker threads and queues are initialized
             for service_id in replication_request.service_ids:
                 if __replication_workers_executor__.get(service_id) is None:
                     __set_up_worker__(service_id)
+                else:
+                    #worker threads were initialized in a previous replciation task, ensure that they are still running, else mark
+                    #the corresponding replication service as one to ignore
```
Comment (cmickeyb) [inline python/pdo/contract/replication.py]: would you wrap the comments at a consistent number of columns (around 100 columns), and there is a spelling error in the comment "replciation"

File: python/pdo/contract/replication.py [hunk 3]
```diff
@@ -82,24 +93,26 @@ def __shutdown_workers__():
...
 
+        # wait for a new task, task is the response object
+        try:
+            response = __pending_replication_tasks_manager_queue__.get(timeout=1.0)
+        except:
+            # check for termination signal
+            if __stop_service__:
+                __shutdown_workers__()
+                logger.info("Exiting Replication manager thread")
+                break
```
Comment (cmickeyb) [inline python/pdo/contract/replication.py]: should the logging statement be before the call to __shutdow_workers__()?

**Comment by prakashngit** [comment]
Adding an issue, will be fixed later.

**Pull request closed without merging**